met-web/src/components/survey/edit/EditForm.tsx

```diff
@@ -1,22 +1,97 @@
-import React, { useContext } from 'react';
+import React, { useEffect, useState } from 'react';
 import { cloneDeep } from 'lodash';
 import { Grid, Stack, TextField } from '@mui/material';
-import { ActionContext } from './ActionContext';
-import { MetLabel, PrimaryButtonOld, SecondaryButtonOld } from 'components/common';
 import { SurveyFormProps } from '../types';
-import { useAppTranslation } from 'hooks';
+import { useAppDispatch, useAppTranslation } from 'hooks';
+import { updateSubmission } from 'services/submissionService';
+import { openNotification } from 'services/notificationService/notificationSlice';
+import { useAsyncValue, useBlocker, useNavigate } from 'react-router-dom';
+import { Engagement } from 'models/engagement';
+import { SurveySubmission } from 'models/surveySubmission';
+import { EmailVerification } from 'models/emailVerification';
+import { Button } from 'components/common/Input';
+import { BodyText } from 'components/common/Typography';
+import { openNotificationModal } from 'services/notificationModalService/notificationModalSlice';
 
 export const EditForm = ({ handleClose }: SurveyFormProps) => {
     const { t: translate } = useAppTranslation();
-    const { handleSubmit, isSubmitting, submission, setSubmission } = useContext(ActionContext);
+    const dispatch = useAppDispatch();
+    const navigate = useNavigate();
+    const [verification, , engagement, initialSubmission] = useAsyncValue() as [
+        EmailVerification,
+        unknown,
+        Engagement,
+        SurveySubmission,
+    ];
+
+    const languagePath = `/${sessionStorage.getItem('languageId')}`;
+
+    const [submission, setSubmission] = useState<SurveySubmission>(initialSubmission);
+
+    const isChanged = JSON.stringify(initialSubmission) !== JSON.stringify(submission);
+
+    const blocker = useBlocker(
+        ({ currentLocation, nextLocation }) => isChanged && nextLocation.pathname !== currentLocation.pathname,
+    );
+    useEffect(() => {
```

Comment:
Would be great to capture a pattern like this in a component for reuse.

Reply:
I was thinking the same! I might create a ticket for it